crates/core/tedge_agent/src/agent.rs

```diff
@@ -267,17 +261,37 @@ impl SmAgent {
             );
         }
 
-        let mut mqtt_errors = mqtt.errors;
+        let request_journal = Journal::open(self.config.journal_path()).await?;
+        let persistance_store = request_journal.clone();
+
+        let mut mqtt = Connection::new(&self.config.mqtt_config).await?;
+        let mqtt_errors = mqtt.errors;
+        let mqtt_input = mqtt.received;
+        let mqtt_output = mqtt.published.clone();
+        let mut mqtt_responses = mqtt.published.clone();
+        let health_topic = self.config.response_topic_health.clone();
+        let errors_topics = self.config.errors_topic.clone();
+
         tokio::spawn(async move {
-            while let Some(error) = mqtt_errors.next().await {
```

Comment:
This didn't need to change, did it? We're not gonna use the journal or anything for processing errors, are we?

Reply:
Okay, now I see that you just moved it to a dedicated named function.